Makefile

```diff
@@ -169,27 +85,108 @@ tidyup:
-	@echo "[+] Done"
-
-	@echo "[+] Making tcping_resolve.gif"
-	@vhs $(TAPE_DIR)tcping_resolve.tape -o $(GIFS_DIR)tcping_resolve.gif
-	@echo "[+] Done"
-
-	@echo "[+] Making tcping_json_pretty.gif"
-	@vhs $(TAPE_DIR)tcping_json_pretty.tape -o $(GIFS_DIR)tcping_json_pretty.gif
-	@echo "[+] Done"
+	@env GOOS=linux CGO_ENABLED=0 go build --ldflags '-s -w -extldflags "-static"' -o tcpingDocker
+
+gifs: $(GIF_ARTIFACTS)
+
+# ==================================================
+# Raw binaries
+# ==================================================
+
+# Output directory
+.PRECIOUS: $(TARGET_DIR)/
+$(TARGET_DIR)/:
+	@mkdir -p $@
+
+# Binary for current platform
+.PRECIOUS: $(TARGET_DIR)/$(BIN_NAME)
+$(TARGET_DIR)/$(BIN_NAME): $(TARGET_DIR)/
+	@echo "[+] Building binary for current platform: $@"
+	@go build -ldflags "-s -w -X main.version=$(VERSION)" -o $@;
+
+# Per-target output directory
+.PRECIOUS: $(TARGET_DIR)/%/
+$(TARGET_DIR)/%/:
+	@mkdir -p $@
+
+# Per-target tcping binary
+.PRECIOUS: $(TARGET_DIR)/%/tcping
+$(TARGET_DIR)/%/tcping: $(TARGET_DIR)/%/
+	@echo "[+] Building binary: $@"
+	@export GOOS=$(word 1, $(subst -, ,$*)); \
+	export GOARCH=$(word 2, $(subst -, ,$*)); \
+	[ $(word 3, $(subst -, ,$*)) = static ] && export CGO_ENABLED=0; \
+	go build -ldflags "-s -w -X main.version=$(VERSION)" -o $@;
+
+# Per-target tcping.exe binary (Windows)
+.PRECIOUS: $(TARGET_DIR)/windows-%/tcping.exe
+$(TARGET_DIR)/windows-%/tcping.exe: $(TARGET_DIR)/windows-%/
+	@echo "[+] Building binary: $@"
+	@export GOARCH=$(word 1, $(subst -, ,$*)); \
+	[ $(word 2, $(subst -, ,$*)) = static ] && export CGO_ENABLED=0; \
+	go build -ldflags "-s -w -X main.version=$(VERSION)" -o $@;
+
+# ==================================================
+# Release outputs
+# ==================================================
+
+# Output directory
+$(OUTPUT_DIR)/:
+	@mkdir -p $@
+
+# .tar.gz archive
+$(OUTPUT_DIR)/tcping-%.tar.gz: $(TARGET_DIR)/%/tcping $(OUTPUT_DIR)/
+	@echo "[+] Compressing binary: $@"
+	@tar -C $$(dirname $<) -czvf $@ tcping >/dev/null
+	@sha256sum $@
+
+# .zip archive (Windows)
+$(OUTPUT_DIR)/tcping-windows-%.zip: $(TARGET_DIR)/windows-%/tcping.exe $(OUTPUT_DIR)/
+	@echo "[+] Compressing binary: $@"
+	@zip -j $@ $< >/dev/null
+	@sha256sum $@
+
+# .deb package (Linux)
+$(OUTPUT_DIR)/tcping-%.deb: $(TARGET_DIR)/linux-%-static/tcping $(OUTPUT_DIR)/
+	@echo "[+] Creating debian package: $@"
+	@PKG_DIR=$$(mktemp -dt make-tcping.XXXXX); \
+	\
+	install -Dm 755 -t $$PKG_DIR/usr/bin/ $<; \
+	\
+	mkdir $$PKG_DIR/DEBIAN; pushd $$PKG_DIR/DEBIAN >/dev/null; \
+	echo "Package: tcping" >>control; \
+	echo "Version: $(VERSION)" >>control; \
+	echo "Section: custom" >>control; \
+	echo "Priority: optional" >>control; \
+	echo "Architecture: $*" >>control; \
+	echo "Essential: no" >>control; \
+	echo "Maintainer: $(MAINTAINER)" >>control; \
+	echo "Description: $(DESCRIPTION)" >>control; \
+	popd >/dev/null; \
+	\
+	dpkg-deb --build $$PKG_DIR $@
```

Comment:
Once `.deb package (Linux)` runs successfully, there’s a `control` file that is created in the current working directory. This also is not cleaned up with `make clean` either.

Reply:
I believe this is related to `pushd` failing, causing it to write `control` into CWD. Now that we are forcing bash this issue should be gone.

Reply:
This was after including the shell. I will give it another shot again and see.

Reply:
yea, it is ok now